devito/types/sparse.py

```diff
@@ -40,11 +40,11 @@ class AbstractSparseFunction(DiscreteFunction):
     _sub_functions = ()
     """SubFunctions encapsulated within this AbstractSparseFunction."""
 
-    __rkwargs__ = DiscreteFunction.__rkwargs__ + ('npoint', 'space_order')
+    __rkwargs__ = DiscreteFunction.__rkwargs__ + ('npoint_global', 'space_order')
 
     def __init_finalize__(self, *args, **kwargs):
         super(AbstractSparseFunction, self).__init_finalize__(*args, **kwargs)
-        self._npoint = kwargs['npoint']
+        self._npoint = kwargs.get('npoint', kwargs.get('npoint_global'))
```

Comment:
it's because of issue #1498 . Same story as shape vs shape_global. Here npoint is the decomposed on. But to reconstruct, we need npoint_global.

I think a potentially good, potentially game-changing question to ask ourselves is the following: why, when we reconstruct, do we even care about carrying around the following attribute:

* shape
* data
* distributor
* ...

In other words, I'm now thinking that only the "main" user-constructed should have this info. All other reconstructions get it via `.function` if they really need to; but most importantly, since reconstructions are performed by the compiler, and since the compiler doesn't clearly care about shape or data, why the hell do we even need all this maddness?

IOW, TLDR: there might be a subset of attributes that are useless at reconstruction time. 

That said, I'm diverging here, perhaps food for thought for another day